pkg/minikube/extract/extract.go

```diff
@@ -0,0 +1,353 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package extract
+
+import (
+	"encoding/json"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/golang-collections/collections/stack"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"k8s.io/minikube/pkg/minikube/console"
+	"k8s.io/minikube/pkg/minikube/exit"
+)
+
+var blacklist = []string{"%s: %v"}
+
+type extractor struct {
```

Comment:
add a comment for the type and fields. It's fairly opaque at this point otherwise.

With this name, the type almost sounds like an interface name rather than a data type name. https://golang.org/doc/effective_go.html#interface-names

I have no suggestion for an alternative name that is meaningful and doesn't stutter, but just food for thought.

Reply:
I've renamed it currentState in an effort to convey what it's actually for. Also added comments.

Reply:
I'm not seeing this change? Anyways, "state" sounds like a good name. "current" is implicit, so I would only use "currentState" if there is a different struct for past state.